rspace/src/main/scala/coop/rchain/rspace/merger/StateMerger.scala

```diff
@@ -0,0 +1,15 @@
+package coop.rchain.rspace.merger
+
+import coop.rchain.rspace.Blake2b256Hash
+import coop.rchain.rspace.trace.{Event, Produce}
+
+import scala.language.higherKinds
+
+trait StateMerger[F[_], C, P, A, K] {
+  def merge(
+      baseState: Blake2b256Hash,
```

Comment:
what do these states mean?

Reply:
I just put comments on the method. Does that comments make sense to you?

Reply:
@zsluedem great! would be good to add also some words about why baseState matters.

Reply:
@zsluedem maybe this trait should also return event logs that should be marked as rejected? Now client have to extract deploys from `mergingState` and make a diff with returned `eventLogs `, is that correct?

Reply:
put more comments now